### DISCUSSION TITLE
Using multiple providers

### DISCUSSION THREAD

**Discussion by jinsley8** [wevm/wagmi#1747] [Q&A]
When I use `alchemyProvider` and `publicProvider` together when configuring the `wagmi` client. It seems to default to public.

Is this the expected behaviour? `publicProvider` seems to cause connection issues and issues with transactions sometimes.

```
const { chains, provider } = configureChains(
  [mainnet,polygon],
  [
    alchemyProvider({ apiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY! }),
    publicProvider(),
  ],
)
```

**Comment by lotinasa** [comment]
In my case stallTimeout is the reason why it gets propagated from alchemy to public. With increased timeout e.g. 2000ms it doesn't reach public provider. Also, when it is using public provider there are a lot of connection issues which are expected as it is public and free, a lot of users are using and reaching its limits.
